Rename rollover crates

luckysori opened the pull request:
The `v_{1,2}_0_0` suffixes are noisy and too verbose. And using the word "deprecated" correctly flags the older version as one that will be phased out soon.

We leverage `pub use current::*` so that we don't have to add `current` as part of any imports.

luckysori committed:
refactor: Rename rollover crates

The `v_{1,2}_0_0` suffixes are noisy and too verbose. And using the
word "deprecated" correctly flags the older version as one that will
be phased out soon.

We leverage `pub use current::*` so that we don't have to add
`current` as part of any imports.

Restioson reviewed (commented):
If a version 3 of rollover happens, would support for v1 be removed?

da-kami commented on maker/src/actor_system.rs:
```diff
@@ -272,8 +268,8 @@ where
                 order,
-                rollover_v_1_0_0_addr.clone(),
-                rollover_v_2_0_0_addr.clone(),
+                rollover_deprecated_addr.clone(),
```
🙃 maybe better to flip these so we always have `rollover_addr` first and the deprecated on second.

da-kami reviewed (approved):
I initially liked naming by version because it was explicit. But this is more intuitive when integrating the protocols, and the fact that we will unlikely support more than two version at the moment justifies this change. Let's do it :)

luckysori committed:
refactor: Remove all noisy v1 protocol suffixes

As stated in 7653d631287642a19d8c48560e74d53dad85d808, these suffixes
are noisy. In the context of the `listen_protocols` module, they will
eventually become a hassle to maintain, because we will have to
constantly update them as the protocol versions change.

luckysori committed:
refactor: Reorder fields based on relevance

It is a bit more intuitive to keep fields that talk about deprecated
protocols below the ones that talk about the newer versions.

Co-authored-by: Daniel Karzel <daniel.karzel@coblox.tech>
